src/prefect/server/orchestration/core_policy.py

```diff
@@ -373,7 +374,17 @@ async def before_transition(
         else:
             delay = base_delay
 
-        if run_settings.retries is not None and run_count <= run_settings.retries:
+        # set by user to conditionally retry a task using @task(retry_condition_fn=...)
+        retriable_set_to_false = bool(
+            proposed_state.state_details.retriable
+            and proposed_state.state_details.retriable is False
+        )
+
+        if (
+            run_settings.retries is not None
+            and run_count <= run_settings.retries
+            and not retriable_set_to_false
+        ):
```

Comment:
I think you might need to conditionally try and pull this with `getattr` because not all state details will have it:

```suggestion
        # set by user to conditionally retry a task using @task(retry_condition_fn=...)
        if getattr(proposed_state.state_details, "retriable", True) is False:
           return
 
        if (
            run_settings.retries is not None
            and run_count <= run_settings.retries
        ):
```

Reply:
The tests passed even when I removed the `retriable` field from the schema(s) but I'd rather be safe than sorry!